elastic/kibana
# [Uptime][Synthetics Integration] Add browser monitors configuration options

## Summary

Relates to elastic/uptime#276 

This PR adds the corresponding UI to support browser monitor configuration within the Synthetics integration package. 

<img width="731" alt="Screen Shot 2021-06-22 at 9 43 11 PM" src="https://user-images.githubusercontent.com/11356435/123024763-9784e400-d3a7-11eb-8878-2b3f78ab75c8.png">
<img width="729" alt="Screen Shot 2021-06-22 at 9 43 04 PM" src="https://user-images.githubusercontent.com/11356435/123024764-981d7a80-d3a7-11eb-8973-83adb656dc03.png">

**Testing**
This feature involves coordinated changes from **kibana**, **integrations**, and **beats**. In order to test e2e, it is necessary to pull down each PR separately. However, there are a few things you can test without every piece.

1. If you only have Kibana
    - Evaluate the design and flow
    - Evaluate that the monitor type dropdown only shows datastreams available from the integration package. To do this, you can change the version in the url to 0.2.1, which only contains http, tcp, and icmp and confirm that the monitor type dropdown only contains http, tcp, and icmp. 
2. If you have Kibana and the integration package
    - Save the integration policy and verify the keys in the integration policy yaml are correct and correspond to the appropriate heartbeat configuration fields. Reference: https://www.elastic.co/guide/en/beats/heartbeat/current/monitor-browser-options.html
    - Edit the integration policy, ensure all the inputs have the appropriate value, change fields, resave the policy, and verify the keys have been updated in the integration policy yaml
3. If you have Kibana, the integration package, and agent
    - Test that browser monitors with configured with a zip url are processed by agent and appear in Uptime
    - Test that browser monitors configured with inline scripts are processed by agent and appear in Uptime

**FULL DEV ENVIRONMENT SETUP INSTRUCTIONS**: https://docs.google.com/document/d/1mS_94T0zRBtdJZVWLWc_B7iswTfkrGJXn9XMyt9JZjY/edit

### Checklist
# Uptime PR Checklist

The following considerations have been addressed

- [x] **Accessibility** has been considered, relevant aria tags and other accessibility features implemented
- [x] **Telemetry** has been added where relevant
- [ ] **Docs** have been added to this PR covering any new, changed, or removed features
- [ ] **Testing** for expected behavior is in place
  - [x] **Automated tests** exist to cover expected and edge case conditions
  - [ ] **User acceptance testing** has been carried out to ensure the feature functions as expected within the context of how it will be used
  - [ ] Any special/edge cases that need to be **manually tested** must be documented
  - [x] Ensure the new layout works **responsively** (including down to small phone widths, where makes sense for the user flow, e.g. the error page when reacting to an alert)
